dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/context/annotation/DubboClassPathBeanDefinitionScanner.java

```diff
@@ -57,14 +64,20 @@ public DubboClassPathBeanDefinitionScanner(BeanDefinitionRegistry registry, Envi
 
     }
 
-    @Override
-    public Set<BeanDefinitionHolder> doScan(String... basePackages) {
-        return super.doScan(basePackages);
-    }
-
     @Override
     public boolean checkCandidate(String beanName, BeanDefinition beanDefinition) throws IllegalStateException {
         return super.checkCandidate(beanName, beanDefinition);
     }
 
+    @Override
+    public Set<BeanDefinition> findCandidateComponents(String basePackage) {
+        Set<BeanDefinition> beanDefinitions = beanDefinitionMap.get(basePackage);
+        // if beanDefinitions size ge 0 => don't scan
+        // if beanDefinitions size is null => scan
```

Comment:
> // if beanDefinitions size ge 0 => don't scan

The comment is inaccurate, it is possible that the scan result is empty

Reply:
> > // if beanDefinitions size ge 0 => don't scan
> 
> The comment is inaccurate, it is possible that the scan result is empty.
> Scan only when the cache is not found.

beanDefinitions size >= 0, means that the scan has been executed.
Because the result of the scan may be equal to 0, only beanDefinitions is null to scan.

Reply:
Please remove redundant comments.